zkvm/src/predicate.rs

```diff
@@ -71,6 +73,39 @@ impl Predicate {
     }
 }
 
+impl PredicateWitness {
+    pub fn to_point(&self) -> CompressedRistretto {
+        self.to_uncompressed_point().compress()
+    }
+
+    fn to_uncompressed_point(&self) -> RistrettoPoint {
+        let gens = PedersenGens::default();
+        match self {
+            PredicateWitness::Key(s) => s * gens.B,
+            PredicateWitness::Or(b) => {
+                let mut t = Transcript::new(b"ZkVM.predicate");
+                let (left, right) = (&b.0.to_uncompressed_point(), &b.1.to_uncompressed_point());
```

Comment:
... and then do:
```suggestion
                let (left, right) = (&l.to_uncompressed_point(), &r.to_uncompressed_point());
```

Reply:
That works if you change `Box<(l,r)>` into `(Box<l>, Box<r>)`

Reply:
which i think fits better with all the code - boxing specific thing, not a group of things.